app/services/rebate_forms_service.rb

```diff
@@ -47,13 +47,24 @@ def create_rebate_form(property)
   end
 
   def create_or_update_property(council)
+    return create_or_update_property_with_valuation_id(council) if @rebate_form_attributes['valuation_id']
+    find_or_create_property_with_no_valuation_id(council)
+  end
+
+  def create_or_update_property_with_valuation_id(council)
     property = Property.find_or_create_by(valuation_id: @rebate_form_attributes['valuation_id'],
                                           rating_year: ENV['YEAR'],
                                           council: council)
     property.update(location: @rebate_form_attributes['location'])
```

Comment:
this should probably use `update!` to raise an exception if the record isn't valid - currently it will return false and silently fail I think.

Reply:
@lamorrison I have added exception raising and error handling to the service and controllers and updated tests. Can you give it another look? Please note that the tests for the rebate_forms_controller will be updated in a separate ticket. Thanks!